ADD BE UDF bitmap_and and bitmap_or

DanyBin opened the pull request:
1.Add BE udf bitmap_and and bitmap_or
 2.Test environment, modify production bitmap

Commit:
bitmap udf

Commit:
bitmap udf

Commit:
bitmap udf

Commit:
bitmap udf

kangkaisen committed:
Make MAX_SCHEDULING_TABLETS and MAX_BALANCING_TABLETS configurable (#2670)

kangkaisen committed:
Decrease transaction_clean_interval_second config (#2673)

kangkaisen committed:
Make bdbje lock timeout configurable (#2676)

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+}
+StringVal BitmapFunctions::bitmap_and(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
+    RoaringBitmap bitmap;
+    if(!src.is_null){
```
```suggestion
    if (!src.is_null) {
```

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+        } else{
+            bitmap.merge(RoaringBitmap ((char*)src.ptr));
+        }
+    }
```
```suggestion
    } else {
        return StringVal::null();
    }
```
If any argument is null, should return null, it will comply with bitand

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+StringVal BitmapFunctions::bitmap_and(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
+    RoaringBitmap bitmap;
+    if(!src.is_null){
+        if(src.len == 0 ){
```
```suggestion
        if (src.len == 0) {
```

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+        if(src.len == 0 ){
+            bitmap.merge(*reinterpret_cast<RoaringBitmap*>(src.ptr));
+        } else{
+            bitmap.merge(RoaringBitmap ((char*)src.ptr));
```
```suggestion
            bitmap.merge(RoaringBitmap((char*)src.ptr));
```

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,51 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
+    if(!src.is_null){
+        if(src.len == 0 ){
+            bitmap.merge(*reinterpret_cast<RoaringBitmap*>(src.ptr));
+        } else{
```
```suggestion
        } else {
```

imay reviewed (commented):
should fix code style

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

DanyBin committed:
Update be/src/exprs/bitmap_function.cpp

Co-Authored-By: ZHAO Chun <buaa.zhaoc@gmail.com>

Commit:
fix code style and fix argument is null, return null

Commit:
fix code style and fix argument is null, return null

Commit:
fix code style and fix argument is null, return null

Commit:
fix code style and fix argument is null, return null

Commit:
fix code style and fix argument is null, return null

kangpinghuang committed:
[Variable] Fix default rowset type variable (#2680)

Fix defaultRowsetType's type description in SessionVariable

Commit:
fix code style

Commit:
fix code style

Commit:
fix code style

Commit:
[Build] fix jemalloc download (#2681)

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,54 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
     return result;
 }
 
+StringVal BitmapFunctions::bitmap_or(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
```
better rename `src` and `dst` to `lhs` and `rhs`.

imay commented on be/src/exprs/bitmap_function.cpp:
```diff
@@ -388,6 +388,54 @@ BigIntVal BitmapFunctions::bitmap_intersect_finalize(FunctionContext* ctx, const
 }
 
+StringVal BitmapFunctions::bitmap_or(FunctionContext* ctx, const StringVal& src, const StringVal& dst){
+    RoaringBitmap bitmap;
```
To make code clear, you can do as follow

```
if (lhs.is_null || rhs.is_null) {
    return StringVal::null();
}
```

And then there is no need to check src.is_null again.

DanyBin replied:
done

DanyBin replied:
done

HangyuanLiu committed:
Support load orc format in Apache Doris (#2554)

Support load orc format in Apache Doris

infearOnTheWay committed:
Fix mysqlChannel npe in ConnectContext (#2684)

kangpinghuang committed:
[Variable] Fix default rowset type read (#2687)

fix bug of read default rowset type in HeartbeatFlags.

kangpinghuang committed:
[Load] Fix broker load's file format npe bug (#2689)

yangzhg committed:
Fix incompatibility with arm architecture in olap #2645 (#2682)

kangkaisen committed:
Change bitmap functions return type to BITMAP (#2690)

Commit:
fix null

Commit:
fix null

lingbin committed:
Remove boost filesystem exception in FileUtils (#2692)

If `errer_code` is provided, then the `boost::filesystem` functions
will not throw an exception, so we do not need to catch it.

Commit:
[Compaction] Fix a bug that CumulativeCompaction compares time of different precision (#2693)

time(NULL) returns second-resolution timestamp, however all compaction related time in Tablet are in millis-resolution. Therefore should use UnixMillis() instead.

morningman committed:
[AlphaRowset] set num_segments field in rowset meta if missing (#2658)

the num segments should be read from rowset meta pb.
But the previous code error caused this value not to be set in some cases.
So when init the rowset meta and find that the num_segments is 0(not set),
we will try to calculate the num segments from AlphaRowsetExtraMetaPB,
and then set the num_segments field.
This should only happen in some rowsets converted from old version.
and for all newly created rowsets, the num_segments field must be set.

caiconghui committed:
[Doc]Remove used doc content about be_rpc_port (#2694)

imay committed:
Update notice year to 2020 (#2696)

kangkaisen commented:
@DanyBin  Hi, refer to https://github.com/apache/incubator-doris/pull/2690/files, please update the   return type to bitmap in gensrc/script/doris_builtins_functions.py

Commit:
return type to bitmap

Commit:
Merge branch 'be_bitmap_udf-0.2' of github.com:DanyBin/incubator-doris into be_bitmap_udf-0.2

DanyBin commented:
ok 

> @DanyBin Hi, refer to https://github.com/apache/incubator-doris/pull/2690/files, please update the return type to bitmap in gensrc/script/doris_builtins_functions.py

ok
